python/pyspark/sql/types.py

```diff
@@ -572,7 +545,7 @@ def __init__(self, startField: Optional[int] = None, endField: Optional[int] = N
         self.startField = startField
         self.endField = endField
 
-    def _str_repr(self) -> str:
+    def typeName(self) -> str:  # type: ignore[override]
```

Comment:
I think this is breaking change to existing class method usage, e.g., `StringType.typeName()`